classes/Product.php

```diff
@@ -4234,7 +4234,7 @@ public static function getQuantity(
         // we don't substract products in cart if the cart is already attached to an order, since stock quantity
         // has already been updated, this is only useful when the order has not yet been created
         if ($cart && empty(Order::getByCartId($cart->id))) {
-            $cartProduct = $cart->getProductQuantity($idProduct, $idProductAttribute, $idCustomization);
+            $cartProduct = $cart->getProductQuantity($idProduct, $idProductAttribute, null);
```

Comment:
Why not take customization into account anymore? Will it not change the returned quantity? Or is it done on purpose because you need the total quantity of the product here regardless of its customization?

Reply:
Yep, it's done on purpose, cause customization should not be taken into account when retrieving the stock quantity. We don't have a different stock per customization.

Reply:
That makes sense, but why is it useful sometimes then? Is it because there are two use cases?
- quantity per row for the product with customization
- the total quantity of the product regardless of its customizations

Reply:
ANother question although I understand the idea, but since `$idCustomization` is a parameter of the method why not use it? its default value is null anyway so using it would actually offer more possibility for the `Product::getQuantity` method:
- if you need the total of the product you use null
- if you want to sub total of a customization you specify it

Systematically ignoring the parameter reduces this method's possibilities, if there's some bug related to this it should be fixed in the code that is calling this method, not forcing a behaviour for everyone and all use cases

Reply:
I think that's exactly what I did. There are 3 scenarios where you want:
- The quantity of the product row with no customization
- The quantity of the product row with the specified customization id
- The total quantity of the product in the cart (with or without customization)

Respectively, you have to provide a customization id equal to:
- 0
- the customization id (e.g: 3)
- null

(The two first scenarios are pretty the same, but it's to emphasize that **0** and **null** scenarios are not the same.)

Here in the product `getQuantity` method, I use null cause we want to get the total quantity in the cart.